### PR TITLE
use -march=x86-64 -mtune=generic instead of -xSSE2 when using Intel oneAPI compilers

### PR DESCRIPTION
**Pull request by boegel**
Required since `xSSE2` is no longer supported by oneAPI compilers:

```
error: unknown target CPU 'SSE2'
note: valid target CPU values are: nocona, core2, penryn, bonnell, atom, silvermont, slm, goldmont, goldmont-plus, tremont, gracemont, nehalem, corei7, westmere, sandybridge, corei7-avx, ivybridge, core-avx-i, haswell, core-avx2, broadwell, common-avx512, skylake, skylake-avx512, skx, cascadelake, cooperlake, cannonlake, icelake-client, rocketlake, icelake-server, tigerlake, sapphirerapids, alderlake, knl, knm, k8, athlon64, athlon-fx, opteron, k8-sse3, athlon64-sse3, opteron-sse3, amdfam10, barcelona, btver1, btver2, bdver1, bdver2, bdver3, bdver4, znver1, znver2, znver3, x86-64, x86-64-v2, x86-64-v3, x86-64-v4
```

tested with HPL in https://github.com/easybuilders/easybuild-easyconfigs/pull/16962 using `eb --from-pr 16962 HPL-2.3-intel-2022b.eb --optarch=GENERIC`

### CODE REVIEW

**Review by migueldiascosta — APPROVED** [review]
lgtm

**Review by bartoldeman — DISMISSED** [review]
LGTM

File: test/framework/toolchain.py
Comment (akesandgren) [inline test/framework/toolchain.py:712]: This is a bit too lax. It should test intel-compilers with and without oneapi enabled.
So perhaps skip the enforcing of oneapi=true below and have two intel-compilers, one pre-2022.2.0 (like the 2021.4.0 above) with -xSSE2 and one intel-compilers using version > 2022.2.0 to get oneapi auto enabled.
Not sure how to do that easy since you can't have two intel-compilers in this dict...

perhaps an array of single compiler dicts?

Comment (boegel) [reply test/framework/toolchain.py]: Very good point for raising this: I've extended this test extensively in 848293d, and doing so actually uncovered a bug, which was fixed in 63099bf

**Review by migueldiascosta — APPROVED** [review]
lgtm, but it looked good to me before, so waiting for an extra pair of eyes before merging

**Comment by migueldiascosta** [comment]
nevermind :)

**Comment by migueldiascosta** [comment]
ah, it was on auto-merge, missed that... even then, I would have expected that a more recent commit would clear the auto-merge, same as dismissing an outdated review...

**Pull request merged**
